### DISCUSSION TITLE
Feature proposal: connection multiplexed pub/sub

### DISCUSSION THREAD

**Discussion by hrvbos** [redis/go-redis#2199] [Ideas]
Hi,

In my company we have a feature proposal that I started working on and would like to know if you are interested in having it in the client.
Following use case is not covered easily by the current client implementation:
  - Pub/Sub with big number of concurrent subscribers each interested in a different channel(think tens of thousands subscribers)
  - Each subscriber listens to only one channel and is usually not a long lasting subscriber(maybe a minute or a few minutes)
  - In the current client implementation new connection will be opened for each subscriber
  - We would like to cover this use case without opening a connection for each subscriber to prevent hitting the limits for number of opened connections

Proposal would be to add connection multiplexing for the pub/sub to the client.
Client would listen to changes for all the channels over a shared connection and handle message delivery to subscribers.
Example of this can be seen in Stack Exchange Redis C# client:
https://github.com/StackExchange/StackExchange.Redis
https://stackexchange.github.io/StackExchange.Redis/

Are you interested in having this in the client?
Would you like me to open a draft PR for the possible implementation?
Just FYI,
I started working on multiplexing pub/sub and it seems to me the easiest way would be to create a wrapper around current PubSub.
So "MultiplexingPubSub" would use current PubSub structure to get all the changes from Redis and then deliver it correctly via mapping.
But, we can discuss the possible implementation in the draft PR in case you are interested.

BR,
Hrvoje

**Comment by gravgaard** [comment]
I would very much like this feature. 

I see from these issues
https://github.com/redis/go-redis/issues/1436
https://github.com/redis/go-redis/issues/785 
that is has been rejected before
